### PR TITLE
Domains: Update outdated domain mapping mentions

### PR DESCRIPTION
**Pull request by leonardost**
#### Changes proposed in this Pull Request

This PR updates some copy where outdated domain mapping information was being displayed. Two places were updated:
- Domain search notices
- Inbound domain transfer

In these places, in some conditions, domain mapping was mentioned as a stand-alone product, which we don't offer anymore. These mentions were removed and improved. Both sign-up and existing site domain flows were updated. Also, the call to action links in these mentions were updated to take the user to the domain mapping flow instead of to the plan selection or checkout pages, hopefully making the flows a little clearer.

This update was prompted by issues reported in p2MSmN-8uJ-p2.

* * *

#### Before

When trying to register a domain with an unsupported TLD:

Screen Shot 2021-07-02 at 22 32 27

After clicking the CTA, the user was taken to the plan selection screen:

Screen Shot 2021-07-02 at 22 32 46

### After

Screen Shot 2021-07-02 at 22 46 27

When clicking the updated CTA, the user is now taken to the domain mapping page, which has more explanations about how mapping works:

Screen Shot 2021-07-02 at 22 46 35

### Before

When trying to add a domain with an unsupported TLD to an existing site:

Screen Shot 2021-07-02 at 22 49 28

When clicking the CTA, the user was taken straight to the checkout page:

Screen Shot 2021-07-02 at 22 35 02

### After

Screen Shot 2021-07-02 at 23 07 19

When clicking the updated CTA, the user is taken to the mapping page:

Screen Shot 2021-07-02 at 22 51 57

## Testing instructions

You can test with the live Calypso link or locally by running this branch in your environment.

Please test both the sign-up flow (access the `/start` URL) and the domains management flow (in Calypso - `Upgrades > Domains`).

Domain search:
* In the domains search bar, input a domain with a TLD we don't offer (e.g. `yourgroovydomain.eu`) and ensure the notice does not mention a price to connect a domain (as we do not offer stand-alone domain mappings anymore).
* Click the CTA link (`connect it`) and ensure it takes you to the domain mapping page

Inbound domain transfer:
* In the inbound domain transfer page, input a domain with a TLD we don't offer (e.g. `yourgroovydomain.eu`) and ensure the text is the updated one:

Screen Shot 2021-07-02 at 23 03 57

* Click on the CTA and ensure you are taken to the domain mapping page.

Screen Shot 2021-07-02 at 23 04 35

### CODE REVIEW

File: client/components/domains/domain-search-results/index.jsx [hunk 1]
```diff
@@ -131,17 +131,17 @@ class DomainSearchResults extends React.Component {
 			) {
 				if ( isDomainMappingFree( selectedSite ) || isNextDomainFree( this.props.cart ) ) {
 					offer = translate(
-						'{{small}}If you purchased %(domain)s elsewhere, you can {{a}}map it{{/a}} for free.{{/small}}',
+						'{{small}}If you purchased %(domain)s elsewhere, you can {{a}}connect it{{/a}} for free.{{/small}}',
```
Comment (a8ci18n) [inline client/components/domains/domain-search-results/index.jsx:133]: :ok: This change will be queued for retranslation. We'll use the existing translations in the meantime.

File: client/components/domains/domain-search-results/index.jsx [hunk 2]
```diff
@@ -131,17 +131,17 @@ class DomainSearchResults extends React.Component {
...
 						{ args: { domain }, components }
 					);
 				} else if ( ! domainsWithPlansOnly ) {
 					offer = translate(
-						'{{small}}If you purchased %(domain)s elsewhere, you can {{a}}map it{{/a}} for %(cost)s.{{/small}}',
-						{ args: { domain, cost: this.props.products.domain_map.cost_display }, components }
+						'{{small}}If you purchased %(domain)s elsewhere, you can {{a}}connect it{{/a}} with WordPress.com Premium.{{/small}}',
```
Comment (a8ci18n) [inline client/components/domains/domain-search-results/index.jsx]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **17** times:
`translate( '{{small}}If you purchased %(domain)s elsewhere, you can {{a}}transfer it{{/a}} with WordPress.com Premium.{{/small}}' )` *ES Score: 7*
See 1 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/update/domain-mapping-mentions)

File: client/components/domains/domain-search-results/index.jsx [hunk 3]
```diff
@@ -131,17 +130,12 @@ class DomainSearchResults extends React.Component {
...
 					);
-				} else if ( ! domainsWithPlansOnly ) {
-					offer = translate(
-						'{{small}}If you purchased %(domain)s elsewhere, you can {{a}}map it{{/a}} for %(cost)s.{{/small}}',
-						{ args: { domain, cost: this.props.products.domain_map.cost_display }, components }
-					);
 				} else {
 					offer = translate(
-						'{{small}}If you purchased %(domain)s elsewhere, you can {{a}}map it{{/a}} with WordPress.com Premium.{{/small}}',
+						'{{small}}If you purchased %(domain)s elsewhere, you can {{a}}connect it{{/a}} with WordPress.com Premium.{{/small}}',
```
Comment (a8ci18n) [inline client/components/domains/domain-search-results/index.jsx:138]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **17** times:
`translate( '{{small}}If you purchased %(domain)s elsewhere, you can {{a}}transfer it{{/a}} with WordPress.com Premium.{{/small}}' )` *ES Score: 7*
See 1 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/update/domain-mapping-mentions)

File: client/my-sites/domains/domain-search/index.jsx [hunk 4]
```diff
@@ -92,13 +92,18 @@ class DomainSearch extends Component {
...
+
+	getDomainMappingUrl = ( domain ) => {
+		let url = '/domains/add/mapping';
+
+		if ( this.props.selectedSiteSlug ) {
+			const query = stringify( { initialQuery: domain.trim() } );
+			url += `/${ this.props.selectedSiteSlug }?${ query }`;
+		}
+
+		return url;
```
Comment (delputnam) [inline client/my-sites/domains/domain-search/index.jsx]: I think we have a function `domainMapping()` in `client/my-sites/domains/paths.js` that will generate this URL for you.

**Review by delputnam — APPROVED** [review]
This LGTM!  ...one comment to look at, but works great. 👍

File: client/my-sites/domains/domain-search/index.jsx (hunk 4 above)
Comment (leonardost) [reply client/my-sites/domains/domain-search/index.jsx]: Good call! I'll reuse that function.

**Comment by leonardost** [comment]
Thanks for reviewing, @delputnam! Updated the code with your suggestion 👍

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/6147537

Thank you @leonardost for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/6147537).

**Pull request merged**